### PR TITLE
feat: streak toggle

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Adds a streak reminder toggle to the popup as well as the dropdown element.
- Converted to it's own component as the actions are the same, and involve quite a lot of logic
- We always subscribe to it, then prompt for push enabling (but don't care if it succeeds or not)

Screenshot 2024-06-12 at 10 09 03
Screenshot 2024-06-12 at 10 08 38

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-367 #done

### Preview domain
https://as-367-streak-toggle.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/pages/account/notifications.tsx [hunk 1]
```diff
@@ -186,7 +186,7 @@ const AccountNotificationsPage = (): ReactElement => {
       logEvent({
         event_name: LogEvent.ScheduleStreakReminder,
         extra: JSON.stringify({
-          hour: readingTimeIndex,
+          hour: 20,
```
Comment (nensidosari) [inline packages/webapp/pages/account/notifications.tsx:189]: why is this changed?

File: packages/shared/src/components/streak/StreakReminderSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    if (value) {
+      logEvent({
+        event_name: LogEvent.ScheduleStreakReminder,
+        extra: JSON.stringify({
+          hour: 20,
+          timezone: user?.timezone,
+        }),
+      });
+      subscribePersonalizedDigest({
+        hour: 20,
```
Comment (nensidosari) [inline packages/shared/src/components/streak/StreakReminderSwitch.tsx:55]: Why is this a hardcoded nr?

**Review by nensidosari — COMMENTED** [review]
Looks good, only 2 things that are somehow related

File: packages/webapp/pages/account/notifications.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/webapp/pages/account/notifications.tsx:189]: It's supposed to be hardcoded 20 was a mistake from the previous implementation as you can't change it.

File: packages/shared/src/components/streak/StreakReminderSwitch.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/streak/StreakReminderSwitch.tsx:55]: Currently we don't support changing this one.
But others are dynamic times, so giving it the ability basically

**Review by nensidosari — APPROVED** [review]
